pkg/gitreceive/build.go

```diff
@@ -138,8 +138,21 @@ func build(
 
 	var pod *api.Pod
 	var buildPodName string
+	image := appName
 	if usingDockerfile {
 		buildPodName = dockerBuilderPodName(appName, gitSha.Short())
+		registryLocation := conf.RegistryLocation
+		registryEnv := make(map[string]string)
+		if registryLocation != "on-cluster" {
+			registryEnv, err = getRegistryDetails(kubeClient, &image, registryLocation, conf.PodNamespace, conf.RegistrySecretPrefix)
+			if err != nil {
+				return fmt.Errorf("error getting private registry details %s", err)
+			}
+			image = image + ":git-" + gitSha.Short()
+		}
+		registryEnv["DEIS_REGISTRY_PROXY_PORT"] = conf.RegistryProxyPort
```

Comment:
> I thought using the DEIS_REGISTRY_SERVICE_PORT will confuse as it is not the actual port on which registry service is running.

But it is, from the perspective of the builder! It just happens to be an nginx instance that forwards requests to the *real* registry. It's not a proxy server. You can interact with it the exact same as you could with the real registry, so I didn't see a reason why we can't re-use the same environment variables already existing.